### ISSUE TITLE
Postgres SimpleArray ends up as TEXT

### ISSUE THREAD

**Issue by rightisleft** [typeorm/typeorm#321]
@Column(ColumnTypes.SIMPLE_ARRAY, {nullable: true}) authors: string[];

```
db=# \d d_t_o_document describe table;
                                       Table "public.d_t_o_document"
    Column     |            Type             |                          Modifiers
---------------+-----------------------------+-------------------------------------------------------------
 id            | integer                     | not null default nextval('d_t_o_document_id_seq'::regclass)
 createdAt     | timestamp without time zone | not null
 modelId       | integer                     |
 documentId    | integer                     | not null
 label         | character varying(255)      | not null
 content       | text                        | not null
 distributions | jsonb                       |
 authors       | text                        |
 title         | text                        |
 source        | text                        |
 month         | text                        |
 year          | text                        |
 day           | text                        |
```

**Comment by icewind** [comment]
As far as I understood it is intentional. To make simple arrays of strings work in all supported databases.
In code there is a comment
```
/**
     * SQL CLOB type. Your class's property type should be array of string. Note: value in this column should not contain
     * a comma (",") since this symbol is used to create a string from the array, using .join(",") operator.
     */
    static SIMPLE_ARRAY: ColumnType = "simple_array";
```
So yes, it is simple text type.

**Comment by rightisleft** [comment]
Odd, would assume https://www.postgresql.org/docs/8.2/static/arrays.html

**Comment by pleerock** [comment]
right, icewind is correct. If lets say if we make array type for postgres specifically then we'll also need to set a type of array. This that it will not be a `simple_array` type already it should be something like `array` plus we need to specify type somewhere, or it can be just `number[]`, `string[]` etc. Postgres will map types directly to arrays, other drivers can map them to text. 

Its an easy feature + its not a priority right now to implement database-specific feature so I'll left it for community to implement.

**Comment by pleerock** [comment]
in latest alpha versions typeorm now supports custom database types including arrays

**Comment by eyalhakim** [comment]
@pleerock how can this be achieved?
when using simple-array with postgres, i still get a text field.

**Comment by pleerock** [comment]
Don't use simple array type. New syntax is as follows:

```ts
@Column("int", { isArray: true })
array: number[];
```

See [examples here](https://github.com/typeorm/typeorm/tree/master/test/functional/database-schema/column-types/postgres).

**Comment by crutchcorn** [comment]
@pleerock shouldn't this be documented in https://typeorm.github.io/tables-and-columns.html?

**Comment by pleerock** [comment]
@crutchcorn yes documentation is outdated, Im going to update it once 0.1.0 is finished and released

**Comment by mskv** [comment]
@pleerock What about migrations? `ColumnSchema` constructor does not accept `isArray` in its options.

**Comment by pleerock** [comment]
@mskv consider using raw sql for now if something is not covered for you. `ColumnSchema` api is not finished yet. But in your case I think you simply need to set type in the ColumnSchema equal to "int[]" for example - it makes same - produces array integer

**Comment by mskv** [comment]
@pleerock 
Thanks for the tip. But for now I've mutated the ColumnSchema after its initialization and all is working :).

```
const someColumnSchema = new ColumnSchema({ name: "some_column", type: "text" });
someColumnSchema.isArray = true;
```
